go/vt/vttablet/tabletserver/query_engine.go

```diff
@@ -236,6 +236,9 @@ func NewQueryEngine(env tabletenv.Env, se *schema.Engine) *QueryEngine {
 
 // Open must be called before sending requests to QueryEngine.
 func (qe *QueryEngine) Open() error {
+	if qe.isOpen {
+		return nil
+	}
```

Comment:
Should this check be not performed under lock or is it protected by higher order?

Reply:
The state manager ensures that Open and Close are not called concurrently. A lock is needed only if the variable is accessed by other functions, which is the case in the other situations.

There are a few places where I do this check inside a lock even though it's not needed. No harm.